src/util/timer.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <chrono>
+
+class timer
+{
+public:
+	explicit timer(std::chrono::nanoseconds delay) :
+	    m_timer(std::chrono::high_resolution_clock::now()),
+		m_delay(std::chrono::duration_cast<std::chrono::high_resolution_clock::duration>(delay))
+	{
+	}
+
+	bool updated()
+	{
+		auto now = std::chrono::high_resolution_clock::now();
+		if ((now.time_since_epoch() - m_timer.time_since_epoch()).count() >= m_delay.count())
```

Comment:
Why the complicated epoch counter nonsense?

```
bool updated() const
{
	auto elapsed_time = std::chrono::high_resolution_clock::now() - m_timer;
	return elapsed_time >= m_delay;
}
```

Reply:
Changed